doxygen/10_UserManual.dox

```diff
@@ -1215,12 +1229,122 @@ calc_kernel_size_func=create_cksf_class(
 Once you have defined <b>how</b> sparse connectivity is going to be initialised, similarly to variables, you can control <b>where</b> it is allocated. 
 This is controlled using the same ``VarLocations`` options described in section \ref sect_var_init_modes and can either be set using the model default specifiued with ``ModelSpec::setDefaultSparseConnectivityLocation`` or on a per-synapse group basis using ``SynapseGroup::setSparseConnectivityLocation``.
 
 -----
-\link sectVariableReferences Previous\endlink | \link UserManual Top\endlink | \link Tutorial1 Next\endlink
+\link sectVariableReferences Previous\endlink | \link UserManual Top\endlink | \link sectToeplitzConnectivityInitialisation Next\endlink
+*/
+//----------------------------------------------------------------------------
+/*! 
+\page sectToeplitzConnectivityInitialisation Toeplitz connectivity initialisation
+
+_Toeplitz matrices_ are ones where the values along all diagonals are constant.
+_Doubly-blocked Toeplitz matrices_ are matrices made out of Toeplitz sub-matrices in a structure that is, itself Toeplitz so the sub-matrices are repeated along diagonals.
+Doubly-blocked Toeplitz matrices can be used to represent common machine learning operations including convolutions and, in GeNN, _Toeplitz connectivity initialisation snippets_ can be used to generate such connectivity on the fly with SynapseMatrixType::TOEPLITZ_KERNELG (see \ref subsect34).
+
+There are a number of predefined Toeplitz connectivity initialisation snippets:
+- InitToeplitzConnectivitySnippet::Conv2D
+- InitToeplitzConnectivitySnippet::AvgPoolConv2D
+
+\add_toggle_python
+In Python, these models can be selected by their unqualified name e.g. "Conv2D".
+\end_toggle
+For example, to initialise convolutional synaptic connectivity with a 3x3 kernel between two populations of 1024 neurons representing a single 32x32 channel:
+\add_toggle_code_cpp
+InitTopelitzConnectivitySnippet::Conv2D::ParamValues conv(
+    3, 3,       // conv_kh, conv_kw
+    32, 32, 1,  // conv_ih, conv_iw, conv_ic
+    32, 32, 1); // conv_oh, conv_ow, conv_oc
+    
+model.addSynapsePopulation<...>(
+        ...
+        initToeplitzConnectivity<InitTopelitzConnectivitySnippet::Conv2D>(conv));
+\end_toggle_code
+\add_toggle_code_python
+conv = {"conv_kh": 3, "conv_kw": 3,
+        "conv_ih": 32, "conv_iw": 32, "conv_ic": 1,
+        "conv_oh": 32, "conv_ow": 32, "conv_oc": 2}
+model.add_synapse_population(
+    ...
+    genn_model.init_toeplitz_connectivity("Conv2D", conv))
+\end_toggle_code
+
+\section sect_new_toeplitz_connect Defining a new Toeplitz connectivity snippet
+Similarly to sparse connectivity initialisation snippets, Toeplitz connectivity initialisation snippets can be created by simply defining a class in the model description.
+
+For example, the following Toeplitz connectivity initialisation snippet could be used to convolve a square kernel with the output of populations of ``kern_size x kern_size`` neurons.
```

Comment:
I am not 100% sure what you mean here. is the kernel``kern_size x kern_size`` or ``sqrt(kern_size) x sqrt(kern_size``?
Why do the populations have ``kern_size x kern_size`` neurons? ``pop_size`` appears to be a parameter below?

Reply:
In trying to simplify this model, I have clearly not helped things :smile: Have renamed the two parameters ``kern_dim`` and ``pop_dim`` and clarified that the kernel is ``kern_dim x kern_dim`` and the population is ``pop_dim x pop_dim``